src/eredis_client.erl

```diff
@@ -402,9 +402,9 @@ maybe_reconnect(Reason, #state{queue = Queue} = State) ->
 reconnect_loop(Client, #state{reconnect_sleep = ReconnectSleep} = State) ->
     case catch(connect(State)) of
         {ok, #state{socket = Socket}} ->
+            Msgs = get_all_messages([]),
```

Comment:
this is a bug in previous version of code, it seems it is trying to read all messages from its own queue, then reconnect then handle the messages in the queue. but in practice, the connection_ready will not be handled before all messages in the queue are rejected.